src/main/java/igym/services/UserService.java

```diff
@@ -22,6 +26,16 @@ public List<User> findAll() {
         return repository.findAll();
     }
 
+    @Transactional
+    public void deleteUser(UUID id) {
+        User user = repository.findById(id).orElseThrow(() -> new UserNotFoundException("User with id " + id + " not found."));
+        if(user.getStatus() == Status.inactive) {  
+            throw new UserAlreadyDeletedException("Gym with id " + id + " is already inactive");
```

Comment:
typo, Should be for a User message. Anyway, take a look in my PR at how we should handle this case,according the comments of @itaayp